client.go

```diff
@@ -136,33 +138,22 @@ func (c *client) request(u string, data interface{}, validator func(body []byte)
 	return nil
 }
 
-// authenticate make a request assuming that the transport has been configured
-// to present the user's token and returns the response from Athenz.
-func (c *client) authenticate() (*AthenzPrincipal, error) {
-	u := fmt.Sprintf("%s/principal", c.endpoint)
-	var ap AthenzPrincipal
-	err := c.request(u, &ap, func(b []byte) error {
-		if ap.Domain == "" || ap.Service == "" {
-			return fmt.Errorf("GET %s unable to get domain and/or name from %s", u, b)
-		}
-		return nil
-	})
-	if err != nil {
-		return nil, err
-	}
-	return &ap, nil
-}
-
-// authorize returns true if the supplied principal has access to the resource and action.
+// authorize returns true if the supplied principal has access to the resource and action. The initial check is done
+// against the zms endpoint. If that is unreachable, the check is retried against the zts endpoint.
 func (c *client) authorize(principal string, check AthenzAccessCheck) (bool, error) {
 	var authzResponse struct {
 		Granted bool `json:"granted"`
 	}
 	esc := url.PathEscape
-	u := fmt.Sprintf("%s/access/%s/%s?principal=%s", c.endpoint, esc(check.Action), esc(check.Resource), esc(principal))
+	u := fmt.Sprintf("%s/access/%s/%s?principal=%s", c.zmsEndpoint, esc(check.Action), esc(check.Resource), esc(principal))
```

Comment:
should we try ZTS first and then fallback to ZMS? ZTS seems to be faster in general compared to ZMS

Reply:
@prabushyam What about the sync period between zms and zts? Do you think this will be a concern if a request comes in before the sync occurs?

Reply:
the sync interval is 1-2 mins, I think it should be ok, unless someone added the access now and trying a kubectl command right after.

Reply:
@prabushyam I have updated it to be zts first then zms.